lib/solr/delete/request.rb

```diff
@@ -1,27 +1,30 @@
 module Solr
   module Delete
     class Request
-      include Solr::ConnectionHelper
-      using Solr::HashExtensions
+      include Solr::Support::ConnectionHelper
+      using Solr::Support::HashExtensions
 
-      PATH = '/update'.freeze
+      SOLR_DELETE_PATH = '/update'.freeze
+
+      attr_reader :core_name, :delete_command
 
       def initialize(options = {})
         options = validate_delete_options!(options)
+        @core_name = options.delete(:core_name)
         @delete_command = { delete: options }
       end
 
       def run(commit: false)
         # need to think how to move out commit data from the connection, it doesn't belong there
-        raw_response = connection(PATH, commit: commit).post_as_json(@delete_command)
+        raw_response = connection(SOLR_DELETE_PATH, core_name: core_name, commit: commit).post_as_json(delete_command)
         Solr::Response.from_raw_response(raw_response)
       end
 
       private
 
       def validate_delete_options!(options)
         options = options.deep_symbolize_keys
-        id, query = options.values_at(:id, :query)
+        core_name, id, query = options.values_at(:core_name, :id, :query)
```

Comment:
why are you getting core_name value and not doing anything with it?

Reply:
I used it to raise an error, but removed that check later